ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala

```diff
@@ -224,15 +253,21 @@ class DomainJsonDecoder(
         .flatMap(jsObj => jsValueToApiValue(lfType, jsObj).flatMap(mustBeApiRecord))
         .valueOr(e => spray.json.deserializationError(e.shows))
 
-  def templateRecordType(id: domain.ContractTypeId.RequiredPkg): JsonError \/ domain.LfType =
+  def templateRecordType(
+      id: domain.ContractTypeId.Template.RequiredPkg
+  ): JsonError \/ domain.LfType =
     resolveTemplateRecordType(id).liftErr(JsonError)
 
-  def keyType(id: domain.ContractTypeId.OptionalPkg)( // TODO #15098 .Template
-      implicit
+  def keyType(id: domain.ContractTypeId.Template.OptionalPkg)(implicit
       ec: ExecutionContext,
       lc: LoggingContextOf[InstanceUUID],
       jwt: Jwt,
       ledgerId: LedgerApiDomain.LedgerId,
   ): ET[domain.LfType] =
-    templateId_(id, jwt, ledgerId).flatMap(it => either(resolveKeyType(it).liftErr(JsonError)))
+    contractTypeId_(id, jwt, ledgerId).flatMap {
+      case it: domain.ContractTypeId.Template.Resolved =>
+        either(resolveKeyType(it: ContractTypeId.Template.Resolved).liftErr(JsonError))
+      case other =>
+        either(-\/(JsonError(s"Expect contract type Id to be template Id, got otherwise: $other")))
+    }
```

Comment:
Did you not mean to just use `templateId_` here and so avoid the manual error checking?